shapeshift/web
# feat: subscribe txs by accountid, multi account support

## Description



we weren't subscribing to txs for account number > 0, this fixes that.

additionally, this revamps the logic to iterate account ids rather than supported chains, eliminating the need to iterate account types as we now have that data in the store as portfolio metadata.


## Notice




## Pull Request Type


## Issue (if applicable)



closes #3043 
closes #2969 

## Risk

high - affects tx subscription logic



## Testing

* connect a wallet - should see `subscribing txs` once with the `accountId`s
* disconnect a wallet - should see `unsubscribing txs` once for all chains the app supports
* switch a wallet - should see `unsubscribing txs` once for all chains the app supports, then
  `subscribing txs` with the new account ids
* enable multi account flag, add an account. you should see unsubscribing, then subscribing with one or more (depending on chain) extra account ids than you previously had


### Engineering

* check the websocket messages look sane in addition to instructions above. check when adding additional we subscribe to multiple addresses. note that the way we call unsubscribe without an address filter will unsubscribe all subscriptions, and we're not able to "oversubscribe"



### Operations

* see common testing instructions



## Screenshots (if applicable)

n/a
